### ISSUE TITLE
[Bug] Release procedure misses steps for updating versions

### ISSUE THREAD

**Issue by wenzeslaus** [OSGeo/grass#2337]
**Describe the bug**

We are missing a guide for creating branches for new major and minor versions and switching versions (on the release branch and main) when new major or minor version is started.

Highly related (at this point same PR), although possibly separate issue, is updating the hardcoded version at various places in the source code.

**Additional context**

Some documentation and artifact updates are needed as well which is tracked in #2336.

The branch creation was originally part of the [howto_release.md](https://github.com/OSGeo/grass/blob/main/doc/howto_release.md) file, however, it got dropped with transition to Git and it is really a separate thing from creating a specific release.

**Comment by wenzeslaus** [comment]
I agree that creating a branch is a separate issue and the version updates are (mostly?) relevant for the minor and major versions (only?). Modified description accordingly.

**Comment by wenzeslaus** [comment]
This needs to happen when branching is done, so it should be addressed in #2963.

**Comment by neteler** [comment]
@wenzeslaus anything left do do here?

**Comment by neteler** [comment]
I take liberty to close this as done.
